src/components/application_manager/src/commands/mobile/add_command_request.cc

```diff
@@ -410,23 +441,43 @@ void AddCommandRequest::on_event(const event_engine::Event& event) {
       application->RemoveCommand(
           (*message_)[strings::msg_params][strings::cmd_id].asUInt());
       result = false;
+      LOG4CXX_DEBUG(logger_, "Result " <<result );
     }
   }
 
   if (BothSend() && hmi_apis::Common_Result::SUCCESS == ui_result_ &&
-      !is_no_vr_error) {
+      !is_no_vr_error &&  hmi_apis::Common_Result::UNSUPPORTED_RESOURCE != vr_result_) {
     result_code = vr_result_ == hmi_apis::Common_Result::REJECTED
                       ? mobile_apis::Result::REJECTED
-                      : mobile_apis::Result::GENERIC_ERROR;
+                      : mobile_apis::Result::GENERIC_ERROR ;
 
     SendHMIRequest(hmi_apis::FunctionID::UI_DeleteCommand, &msg_params);
 
     application->RemoveCommand(
         (*message_)[strings::msg_params][strings::cmd_id].asUInt());
     result = false;
+    LOG4CXX_DEBUG(logger_, "Result " <<result );
   }
 
-  SendResponse(result, result_code, NULL, &(message[strings::msg_params]));
+  if (!BothSend() && is_vr_or_ui_unsupported) {
+      LOG4CXX_DEBUG(logger_, "!BothSend() && is_vr_or_ui_unsupported");
+      result = false;
+  }
+
+  if (is_vr_and_ui_unsupported) {
+      LOG4CXX_DEBUG(logger_, "is_vr_and_ui_unsupported");
```

Comment:
@LuxoftAKutsan, such log message is not very useful. Add more info and interpretation - like == true, or more words.